cmd/expand-pool.go

```diff
@@ -0,0 +1,561 @@
+		pe := selfPoolExpand{}
+		body, err := os.ReadFile(path.Join(lp, minioMetaBucket, poolExpandStatusFile))
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				continue
+			}
+			return nil, err
+		}
+		err = json.Unmarshal(body, &pe)
+		if err != nil {
+			return nil, err
+		}
+		if filemd5 == "" {
+			filemd5 = pe.FileMD5
+		}
+		if pe.FileMD5 != filemd5 {
+			return nil, fmt.Errorf("inconsistent pool expand status")
+		}
+		if poolIndex == -1 {
+			poolIndex = pe.PoolIndex
+		}
+		if pe.PoolIndex != poolIndex {
+			return nil, fmt.Errorf("inconsistent pool expand status")
+		}
+		// if empty, set to PoolExpandStatusComplete
+		// but why it's empty
+		if pe.Status == "" {
+			pe.Status = PoolExpandStatusComplete
+		}
+		statsMap[pe.Status]++
+	}
+	stat := ""
+	maxCount := -1
+	for s, c := range statsMap {
+		if c >= maxCount {
+			stat = string(s)
+			maxCount = c
+		}
+	}
+	if maxCount != -1 && maxCount < len(localPaths)/2 {
+		return nil, fmt.Errorf("pool expand status is not complete, please check the status of the pool expand")
+	}
+	return &selfPoolExpand{
+		Status:    PoolExpandStatus(stat),
+		PoolIndex: poolIndex,
+		FileMD5:   filemd5,
+	}, nil
+}
+
+func getConfigFileMd5() (string, error) {
+	if !globalServerCtxt.EnableExpandPools {
+		return "", fmt.Errorf("pool expand is not enabled")
+	}
+	fileBody, err := os.ReadFile(globalServerCtxt.ConfigPath)
+	if err != nil {
+		return "", err
+	}
+	m5 := md5.New()
+	m5.Write(fileBody)
+	return hex.EncodeToString(m5.Sum(nil)), nil
+}
+
+func writePoolExpandStats(poolExpandStatus *selfPoolExpand) error {
+	localPaths := globalEndpoints.LocalDisksPaths()
+	if poolExpandStatus.Status == "" || poolExpandStatus.FileMD5 == "" {
+		return fmt.Errorf("invalid pool expand status")
+	}
+	body, err := json.Marshal(poolExpandStatus)
+	if err != nil {
+		return err
+	}
+	errCount := 0
+	for _, lp := range localPaths {
+		filePath := path.Join(lp, minioMetaBucket, poolExpandStatusFile)
+		err = os.WriteFile(filePath, body, 0644)
+		if err != nil {
+			errCount++
+		}
+	}
+	if errCount > len(localPaths)/2 {
+		return fmt.Errorf("failed to write pool expand status")
+	}
+	return nil
+}
+
+func initExpandPool(configFile string, ctxt *serverCtxt) error {
+	if !ctxt.EnableExpandPools {
+		return nil
+	}
+	stat, err := os.Stat(configFile)
+	if err != nil {
+		return err
+	}
+	ltime := stat.ModTime()
+	// load expand data and dosomething
+	go func() {
+		ticker := time.NewTicker(time.Second * 10)
+		for {
+			select {
+			case <-ticker.C:
+				stats, err := loadPoolExpandStats()
+				if err != nil {
+					log.Error("loadPoolExpandStats error:", err)
+					continue
+				}
+				switch stats.Status {
+				case "":
+					continue
+				case PoolExpandStatusWaitForFileChange:
+					for index := range globalNotificationSys.peerClients {
+						if globalNotificationSys.peerClients[index] == nil {
+							continue
+						}
+					}
+				case PoolExpandStatusWaitForPreGlobalEndpointsChange:
```

Comment:
Do we plan to add individual actions of these cases or we can just use `default` here?